### PR TITLE
fix(core): fix crash when executing sql. (#327)

### LINKED ISSUE #327: bug: stonedb crashes when executing "SELECT" :
### Describe the problem
```
CREATE TABLE t1 (a INT NOT NULL, b INT)engine=stonedb;
INSERT INTO t1 VALUES (1, 1);
SELECT * FROM t1 WHERE (a=a AND a=a);
##The error results are as follows:
ERROR 2013 (HY000): Lost connection to MySQL server during query
```
mysqld crashes:
```
Signal: SIGSEGV (Segmentation fault)
Process finished with exit code 2
```
### Expected behavior

##The following is：innodb test result
```
SELECT * FROM t1 WHERE (a=a AND a=a);
+---+------+
| a | b    |
+---+------+
| 1 |    1 |
+---+------+
1 row in set (0.00 sec)
```
### How To Reproduce
```
CREATE TABLE t1 (a INT NOT NULL, b INT)engine=stonedb;
INSERT INTO t1 VALUES (1, 1);
SELECT * FROM t1 WHERE (a=a AND a=a);
```
### Environment
```
1. Server OS [Linux]
2. StoneDB Version [5.7]

```
### Are you interested in submitting a PR to solve the problem?

- [X] Yes, I will!

### PR DESCRIPTION
**Pull request by andyli029**
## Summary about this PR

Issue Number: close #327

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by andyli029** [comment]
Fix the crash: #327
TODO：match the result from innodb:

The following is：innodb test result
```
mysql>  SELECT * FROM t1 WHERE (a=a AND a=a);
+---+------+
| a | b    |
+---+------+
| 1 |    1 |
+---+------+
1 row in set (0.00 sec)
```

Result from fixed:
```
mysql> $ SELECT * FROM t1 WHERE (a=a AND a=a);
ERROR 6 (HY000): The query includes syntax that is not supported by the storage engine. Either restructure the query with supported syntax, or enable the MySQL core::Query Path in config file to execute the query with reduced performance.
```

**Review by hustjieke — APPROVED** [review]
Nice job!
LGTM

**Review by isredstar — APPROVED** [review]
Good job

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@ build/
 cscope*
 sql/build_info.h
 .vscode
+.idea
```
Comment (xiaoguangye) [inline .gitignore:5]: gitignore should be ignored?

**Pull request merged**